### PR TITLE
Refactor SliceTupleAndList utility code

### PR DESCRIPTION
**Pull request by jdemeyer**
Factor out the logic for creating a `tuple` and `list` from a C array into new functions `__Pyx_PyTuple_FromArray` and `__Pyx_PyList_FromArray`. This is useful for #3021 for constructing an `*args` tuple in case that that `METH_FASTCALL` is used.

As additional proof that this is useful, CPython also has an internal function `_PyTuple_FromArray` with the same functionality.

### CODE REVIEW

File: Cython/Utility/ObjectHandling.c [hunk 1]
```diff
@@ -735,17 +735,67 @@ static CYTHON_INLINE int __Pyx_PyObject_SetSlice(PyObject* obj, PyObject* value,
...
+}
+#endif
+
+
 /////////////// SliceTupleAndList.proto ///////////////
 
 #if CYTHON_COMPILING_IN_CPYTHON
-static CYTHON_INLINE PyObject* __Pyx_PyList_GetSlice(PyObject* src, Py_ssize_t start, Py_ssize_t stop);
 static CYTHON_INLINE PyObject* __Pyx_PyTuple_GetSlice(PyObject* src, Py_ssize_t start, Py_ssize_t stop);
+static CYTHON_INLINE PyObject* __Pyx_PyList_GetSlice(PyObject* src, Py_ssize_t start, Py_ssize_t stop);
```
Comment (scoder) [inline Cython/Utility/ObjectHandling.c]: Unnecessary change?

Comment (jdemeyer) [reply Cython/Utility/ObjectHandling.c]: Fixed.

**Comment by jdemeyer** [comment]
> I'd be happy to merge #3021 directly instead.

On the other hand, this PR exists now, so why not merge it right away? I find it slightly more convenient to have this part out of the diff for #3021.

**Comment by scoder** [comment]
Thanks
